core/payment-driver/base/src/bus.rs

```diff
@@ -0,0 +1,120 @@
+    let bus_id = driver_bus_id(driver.get_name());
+
+    /* Short variable names explained:
+        db = DbExecutor || ()
+        dr = Arc<dyn PaymentDriver>
+        c = caller
+        m = message
+    */
+    #[rustfmt::skip] // Keep move's neatly alligned
+    ServiceBinder::new(&bus_id, &(), driver)
+        .bind_with_processor(
+            move |db, dr, c, m| async move { dr.init(db, c, m).await }
+        )
+        .bind_with_processor(
+            move |db, dr, c, m| async move { dr.account_event(db, c, m).await }
+        )
+        .bind_with_processor(
+            move |db, dr, c, m| async move { dr.get_account_balance(db, c, m).await }
+        )
+        .bind_with_processor(
+            move |db, dr, c, m| async move { dr.get_transaction_balance(db, c, m).await }
+        )
+        .bind_with_processor(
+            move |db, dr, c, m| async move { dr.schedule_payment(db, c, m).await }
+        )
+        .bind_with_processor(
+            move |db, dr, c, m| async move { dr.verify_payment(db, c, m).await }
+        );
+
+    log::debug!("Successfully bound payment driver service to service bus");
+    log::debug!("Subscribing to identity events");
+    let message = identity::Subscribe { endpoint: bus_id };
+    let result = service(identity::BUS_ID).send(message).await;
+    match result {
+        Err(e) => log::error!("init app-key listener error: {}", e),
+        _ => log::debug!("Successfully subscribed payment driver service to identity events"),
+    }
+}
+
+pub async fn register_account(
+    driver: &(dyn PaymentDriver),
+    address: &str,
+    mode: AccountMode,
+) -> Result<(), GenericError> {
+    let address = address.to_string();
+    let msg = payment_srv::RegisterAccount {
+        platform: driver.get_platform(),
+        address,
+        driver: driver.get_name(),
+        mode,
+    };
+    service(payment_srv::BUS_ID)
+        .send(msg)
+        .await
+        .map_err(GenericError::new)?
+        .map_err(GenericError::new)?;
+    Ok(())
+}
+
+pub async fn sign(node_id: NodeId, payload: Vec<u8>) -> Result<Vec<u8>, GenericError> {
+    let signature = service(identity::BUS_ID)
+        .send(identity::Sign { node_id, payload })
+        .await
+        .map_err(GenericError::new)?
+        .map_err(GenericError::new)?;
+    Ok(signature)
+}
+
+pub fn notify_payment(
+    driver: &(dyn PaymentDriver),
+    order_id: &str,
+    details: &PaymentDetails,
+    confirmation: Vec<u8>,
+) {
+    let msg = payment_srv::NotifyPayment {
+        driver: driver.get_name(),
+        amount: details.amount.clone(),
+        sender: details.sender.clone(),
+        recipient: details.recipient.clone(),
+        order_ids: vec![order_id.to_string()],
+        confirmation: PaymentConfirmation { confirmation },
+    };
+
+    // Spawned because calling payment service while handling a call from payment service
+    // would result in a deadlock.
+    Arbiter::spawn(async move {
```

Comment:
That spawn is only necessary if `notify_payment` is called from `schedule_payment`. I believe that won't be the case if there's a separate thread with send loop.

Reply:
good catch, will update